In [15]:
import numpy as np
from scipy import stats

## `Q.1`. Given a normal distribution with mean as 50 and deviation as 10, answer below questions

(X ~ N(5O, 10) ): 

### a. what are the values of the mean and standard deviation? 

In [1]:
mean = 50
std = 10

### b. What value of x has a z-score of 1.4? 

In [2]:
z = 1.4

In [3]:
# z = (x-mean)/std
x = (z*std) + mean
x

64.0

### c. What is the Z-score that corresponds to x = 30? 

In [4]:
x = 30

In [5]:
z = (x - mean)/std
z

-2.0

### d. What is the difference between positive and negative z values?

The sign of the z-score indicates whether the raw score (x) is above or below the mean of the distribution.


- A positive z-score indicates that the raw score is above the mean, while 


- a negative z-score indicates that the raw score is below the mean. 



- The absolute value of the z-score represents the number of standard deviations away from the mean.

## `Q.2:`The average test score in a certain statistics class was 74 with a standard deviation of 8. There are 2000 students in this class. Use the emperical rule to answer the following questions: 

### (a) What percentage of students scored less than 58? 

In [120]:
n = 2000


In [121]:
std = 8
mean = 74

In [10]:
print(f'first std dev = [{mean - std}, {mean + std}]')
print(f'second std dev = [{mean - 2*std}, {mean + 2*std}]')
print(f'third std dev = [{mean - 3*std}, {mean + 3*std}]')

first std dev = [66, 82]
second std dev = [58, 90]
third std dev = [50, 98]


In [14]:
# less than 58 is 

(100 - 95)/2

2.5

### (b) What is the probability that a student score between 66 and 82 on the exam? 

In [17]:
z1 = (66 - mean)/std
z2 = (82 - mean)/std
z1,z2

(-1.0, 1.0)

In [19]:
stats.norm.cdf(z2)-stats.norm.cdf(z1)

0.6826894921370859

#### There is 68% probability that a student score between 66 and 82 on the exam

### (c) <span class="girk">How many students scored at most 90?</span> 

In [108]:
z = (90-mean)/std
z

2.0

#### 90 is in 2nd std dev ie. 95%:

In [113]:
print(f'percentage of students getting at most 90 is {stats.norm.cdf(z)*100:.3f} which is {np.round(stats.norm.cdf(z)*n)}')

percentage of students getting at most 90 is 97.725 which is 1954.0


### (d) What percentage of students scored at least 66? 

In [133]:
z = (66-mean)/std
z

-1.0

In [134]:
print(f'percentage of students getting scored at least 66 is {(1-stats.norm.cdf(z))*100:.3f} which is {np.round((1-stats.norm.cdf(z))*n)}')

percentage of students getting scored at least 66 is 84.134 which is 1683.0


### (e) How many students scored more than 98 on the test?

In [135]:
z = (98-mean)/std
z

3.0

In [137]:
print(f'percentage of students getting scored more than 98 is {(1-stats.norm.cdf(z))*100:.3f} which is {np.round((1-stats.norm.cdf(z))*n)}')

percentage of students getting scored more than 98 is 0.135 which is 3.0


## `Q-3:` Normally distributed IQ scores have a mean of 100 and a standard deviation of 15. Use the standard z-table to answer the following questions:

What is the probability of randomly selecting someone with an IQ score that is 

###    (a) less than 80? 

In [79]:
mean = 100
std_dev = 15

In [53]:
z = (80-mean)/std_dev
z

-1.3333333333333333

In [54]:
print(f'the probability of a randomly selected person having an IQ score less than 80 is approximately {stats.norm.cdf(z)*100:.2f}%')

the probability of a randomly selected person having an IQ score less than 80 is approximately 9.12%


### (b) greater than 136? 

In [55]:
z = (136-mean)/std_dev
z

2.4

In [57]:
print(f'the probability of a randomly selected person having an IQ score greater than 136 is approximately {(1-stats.norm.cdf(z))*100:.2f}%')

the probability of a randomly selected person having an IQ score greater than 136 is approximately 0.82%


###    (c) between 95 and 110?

In [62]:
z1 = (110-mean)/std_dev
z1

0.6666666666666666

In [63]:
z2 = (95-mean)/std_dev
z2

-0.3333333333333333

In [67]:
print(f'the probability of a randomly selected person having an IQ score between 95 and 110 is approximately {((stats.norm.cdf(z1))*100 -(stats.norm.cdf(z2))*100) :.2f}%')

the probability of a randomly selected person having an IQ score between 95 and 110 is approximately 37.81%


###    (d) What IQ score corresponds to the 90th percentile? 

In [87]:
mean = 100
std = 15

In [88]:
percentile = 0.90

##### using `norm.ppf(percentile)` to calculate z_score of a percentile.

In [89]:
z_score = stats.norm.ppf(percentile)
z_score

1.2815515655446004

In [90]:
mean + (z_score * std_dev)

119.22327348316901

##### The IQ score corresponds to the 90th percentile is 119.22

###    (e) The middle 30% of IQs fall between what two values?

In [92]:
import scipy.stats as stats

# Define the mean and standard deviation
mean = 100
std_dev = 15

# Define the lower and upper percentiles
lower_percentile = 0.35
upper_percentile = 0.85

# Calculate the z-scores for the lower and upper percentiles
z1 = stats.norm.ppf(lower_percentile)
z2 = stats.norm.ppf(upper_percentile)

# Calculate the corresponding IQ scores
iq_score1 = mean + z1 * std_dev
iq_score2 = mean + z2 * std_dev

print(f"The middle 30% of IQs fall between {iq_score1:.2f} and {iq_score2:.2f}")


The middle 30% of IQs fall between 94.22 and 115.55


In [94]:
percentage = 30

In [95]:
z_score = stats.norm.ppf(percentile)
z_score

1.2815515655446004

## `Q-4`: Wine data
```
# Load dataset
import pandas as pd
from sklearn.datasets import load_wine

# load the wine dataset
data = load_wine()
df = pd.DataFrame(data.data, columns=data.feature_names)
df["target"] = data.target

```
Consider this "wine" dataset which can be loaded using above code. It has 13 numerical features and a target column with 3 class: 0, 1, 2. 


* **Which feature is the best predictor of the wine class in the wine dataset?** (Using pdf plot)

* **Which two combination of features from below listed features are best for wine classification?** (2D pdf plot)
    * Proline
    * Flavanoids
    * Color intensity
    * OD280/OD315 of diluted wines
    * Alcohol

In [ ]:
# code here